test/prototype_transforms_kernel_infos.py

```diff
@@ -2100,3 +2118,39 @@ def sample_inputs_convert_dtype_video():
         ),
     ]
 )
+
+
+def sample_inputs_uniform_temporal_subsample_video():
+    for video_loader in make_video_loaders(sizes=["random"], num_frames=[4]):
+        for temporal_dim in [-4, len(video_loader.shape) - 4]:
+            yield ArgsKwargs(video_loader, num_samples=2, temporal_dim=temporal_dim)
+
+
+def reference_uniform_temporal_subsample_video(video, num_samples, temporal_dim=-4):
+    from pytorchvideo.transforms.functional import uniform_temporal_subsample
+
+    return uniform_temporal_subsample(video, num_samples=num_samples, temporal_dim=temporal_dim)
+
+
+def reference_inputs_uniform_temporal_subsample_video():
+    for video_loader in make_video_loaders(sizes=["random"], color_spaces=[features.ColorSpace.RGB], num_frames=[10]):
+        for num_frames in range(1, 11):
+            yield ArgsKwargs(video_loader, num_frames)
+
+
+KERNEL_INFOS.append(
+    KernelInfo(
+        F.uniform_temporal_subsample_video,
+        sample_inputs_fn=sample_inputs_uniform_temporal_subsample_video,
+        reference_fn=reference_uniform_temporal_subsample_video,
+        reference_inputs_fn=reference_inputs_uniform_temporal_subsample_video,
+        test_marks=[
+            skip_on_missing_third_party_reference("pytorchvideo"),
+            TestMark(
+                ("TestKernels", "test_batched_vs_single"),
+                pytest.mark.skip("Positive `temporal_dim` arguments are not equivalent for batched and single inputs"),
+                condition=lambda args_kwargs: args_kwargs.kwargs.get("temporal_dim") >= 0,
+            ),
+        ],
+    )
+)
```

Comment:
Finally, we need to bring everything above together. The information is bundled inside a `KernelInfo` and appended to long `KERNEL_INFOS` list that holds all defined `KernelInfo`'s in this file. The only thing new here are the `test_marks`. This field lets us conditionally apply `pytest.mark.*`'s to specific tests. There are two defined here:

1. If `pytorchvideo` is not available, we want to skip the reference test. Otherwise, calling `reference_uniform_temporal_subsample_video` will fail. Instead of doing this through the `test_marks`, we could have also used `pytorchvideo = pytest.importorskip("pytorchvideo")` inside the reference function. That has ultimately the same effect since the test will be skipped, but has a major disadvantage here: the test will only be skipped when the reference function is called. Looking at the implementation of the test

   https://github.com/pytorch/vision/blob/9c112935abe400222cca8f9fbc2d8386e0f25e80/test/test_prototype_transforms_functional.py#L193-L197

   at this point we have already instantiated the inputs as well as already called our kernel. Thus, compared to just skipping before the test is run in the first place, we are wasting resources here.

2. The `test_batched_vs_single` tests whether the output is identical for batched inputs or the same inputs, but with unbound batch dimensions. This works fine here if we use negative indices for `temporal_dim` since the "data dimensions" 

   https://github.com/pytorch/vision/blob/9c112935abe400222cca8f9fbc2d8386e0f25e80/test/test_prototype_transforms_functional.py#L126-L137

   will not be touched by the test. However, using positive indices is not possible, since that would need to be fixed after unbatching. Thus, we skip this test conditionally in case we encounter positive indices.